release-notes/opensearch-sql.release-notes-3.1.0.0.md

```diff
@@ -0,0 +1,57 @@
+* Support `top`, `rare` commands with Calcite ([#3647](https://github.com/opensearch-project/sql/pull/3647))
+* Add earliest and latest in condition function ([#3640](https://github.com/opensearch-project/sql/pull/3640))
+* Support `fillnull` command with Calcite ([#3634](https://github.com/opensearch-project/sql/pull/3634))
+* Support function `coalesce` with Calcite ([#3628](https://github.com/opensearch-project/sql/pull/3628))
+* Support functions `isempty`, `isblank` and `ispresent` with Calcite ([#3627](https://github.com/opensearch-project/sql/pull/3627))
+* Support `describe` command with Calcite ([#3624](https://github.com/opensearch-project/sql/pull/3624))
+* Support Limit pushdown ([#3615](https://github.com/opensearch-project/sql/pull/3615))
+* Add UT for PredicateAnalyzer and AggregateAnalyzer ([#3612](https://github.com/opensearch-project/sql/pull/3612))
+* Add a new row count estimation mechanism for CalciteIndexScan ([#3605](https://github.com/opensearch-project/sql/pull/3605))
+* Implement `geoip` udf with Calcite ([#3604](https://github.com/opensearch-project/sql/pull/3604))
+* Implement `cidrmatch` udf with Calcite ([#3603](https://github.com/opensearch-project/sql/pull/3603))
+* Support `eventstats` command with Calcite ([#3585](https://github.com/opensearch-project/sql/pull/3585))
+* Add lambda function and array related functions ([#3584](https://github.com/opensearch-project/sql/pull/3584))
+* Implement cryptographic hash UDFs ([#3574](https://github.com/opensearch-project/sql/pull/3574))
+* Calcite patterns command brain pattern method ([#3570](https://github.com/opensearch-project/sql/pull/3570))
+* add json functions ([#3559](https://github.com/opensearch-project/sql/pull/3559))
+
+### Bug Fixes
+* Fix error when pushing down script filter with struct field ([#3693](https://github.com/opensearch-project/sql/pull/3693))
+* Fix alias type referring to nested field ([#3674](https://github.com/opensearch-project/sql/pull/3674))
+* Fix: Long IN-lists causes crash ([#3660](https://github.com/opensearch-project/sql/pull/3660))
+* Add a trimmed project before aggregate to avoid NPE in Calcite ([#3621](https://github.com/opensearch-project/sql/pull/3621))
+* Fix field not found issue in join output when column names are ambiguous ([#3760](https://github.com/opensearch-project/sql/pull/3760))
+* Fix: correct ATAN(x, y) and CONV(x, a, b) functions bug ([#3748](https://github.com/opensearch-project/sql/pull/3748))
+* Return double with correct precision for `UNIX_TIMESTAMP` ([#3679](https://github.com/opensearch-project/sql/pull/3679))
+* Prevent push down limit with offset reach maxResultWindow ([#3713](https://github.com/opensearch-project/sql/pull/3713))
+* Fix pushing down filter with nested filed of the text type ([#3645](https://github.com/opensearch-project/sql/pull/3645))
+* Make query.size_limit only affect the final results ([#3623](https://github.com/opensearch-project/sql/pull/3623))
```

Comment:
maybe we should treat https://github.com/opensearch-project/sql/pull/3623 as a breaking:
`query.size_limit` has been misused as maximum amount of for fetching documents from OpenSearch, causing many data correctness issues. It has now been fixed to the maximum number of rows output of query.

Reply:
Because that, the "maximum amount of fetching documents from OpenSearch" is unlimited. (actually value is `Integer.MAX_VALUE`, requestedTotalSize=2147483647). Query on big data set may fail with CircuitBreakingException. ref #3779